### PR TITLE
Improve javascript

### PR DESCRIPTION
**Pull request by ximex**
- move all js code into js files (except some lines for the `design` page)
- simplify some scripts
- remove type attribute

we could set the `defer` attribute on the `<script>` then we don't need the `ready` function any more. or should we use `async` (then we still need the `ready` function and/or set `type="module"`. if we still use `ready` then we should maybe put all(?) into one `ready` function.
see the following links to understand:
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#attributes
- https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState
- https://gist.github.com/jakub-g/385ee6b41085303a53ad92c7c8afd7a6

if we use `type="module"` maybe we don't need the javascript build process any more.

with this PR we should be done with the js improvements/refactorings

### CODE REVIEW

**Comment by ximex** [comment]
@MyIgel @weeman1337 ready for review/merge

pls write your opinion on the `type="module"` and the `ready()` function thing

File: includes/pages/user_shifts.php [hunk 1]
```diff
@@ -374,20 +368,23 @@ function get_ids_from_array($array)
  * @param array  $selected
  * @param string $name
  * @param string $title
- * @param array  $additionalButtons
+ * @param int[]  $ownAngelTypes
  * @return string
  */
-function make_select($items, $selected, $name, $title = null, $additionalButtons = [])
+function make_select($items, $selected, $name, $title = null, $ownAngelTypes = [])
```
Comment (MyIgel) [inline includes/pages/user_shifts.php:374]: As this is a generic function imho the wording should be kept generic so in that case eventually `$ownSelect`

**Comment by ximex** [comment]
changed

File: includes/pages/user_shifts.php (hunk 1 above)
Comment (ximex) [reply includes/pages/user_shifts.php]: fixed

**Comment by weeman1337** [comment]
Can you solve the conflicts @ximex ?

**Comment by ximex** [comment]
@weeman1337 resolved

**Review by weeman1337 — CHANGES_REQUESTED** [review]
@ximex can you fix the PHP code style?
Looks good otherwise.

```
$ ./vendor/bin/phpcs -p --no-colors --basepath="$PWD"
............................................................  60 / 537 (11%)
..........................E................................. 120 / 537 (22%)
....E....................................................... 180 / 537 (34%)
............................................................ 240 / 537 (45%)
..............................E............................. 300 / 537 (56%)
............................................................ 360 / 537 (67%)
............................................................ 420 / 537 (78%)
............................................................ 480 / 537 (89%)
.........................................................    537 / 537 (100%)
FILE: includes/pages/user_shifts.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 387 | ERROR | [x] Multi-line arrays must have a trailing comma after
     |       |     the last element.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
FILE: includes/view/UserDriverLicenses_view.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 53 | ERROR | [x] Multi-line arrays must have a trailing comma after
    |       |     the last element.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
FILE: src/Controllers/DesignController.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 56 | ERROR | [x] String "Y-m-d" does not require double quotes; use
    |       |     single quotes instead
 57 | ERROR | [x] String "Y-m-d" does not require double quotes; use
    |       |     single quotes instead
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
Time: 11.15 secs; Memory: 30MB
Cleaning up project directory and file based variables 00:01
ERROR: Job failed: exit code 1
```

**Comment by ximex** [comment]
@weeman1337 done

**Comment by weeman1337** [comment]
Still some PHPCS errors @ximex :

```
$ ./vendor/bin/phpcs -p --no-colors --basepath="$PWD"
............................................................  60 / 537 (11%)
..........................E................................. 120 / 537 (22%)
....E....................................................... 180 / 537 (34%)
............................................................ 240 / 537 (45%)
..............................E............................. 300 / 537 (56%)
............................................................ 360 / 537 (67%)
............................................................ 420 / 537 (78%)
............................................................ 480 / 537 (89%)
.........................................................    537 / 537 (100%)
FILE: includes/pages/user_shifts.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 387 | ERROR | [x] Multi-line arrays must have a trailing comma after
     |       |     the last element.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
FILE: includes/view/UserDriverLicenses_view.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 53 | ERROR | [x] Multi-line arrays must have a trailing comma after
    |       |     the last element.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
FILE: src/Controllers/DesignController.php
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 56 | ERROR | [x] String "Y-m-d" does not require double quotes; use
    |       |     single quotes instead
 57 | ERROR | [x] String "Y-m-d" does not require double quotes; use
    |       |     single quotes instead
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
Time: 9.33 secs; Memory: 30MB
```

**Review by weeman1337 — APPROVED** [review]
Thanks :+1:

**Pull request merged**